### PR TITLE
Dmsghttp Whitelist

### PR DESCRIPTION
**Pull request by 0pcom**
implementation of a dmsghttp fileserver which uses a whitelist 
* if no keys are whitelisted, the server is publicly accessible

Help menu
image

Request logging
image

Request from whitelisted PK with 200 response
image

Request from non-whitelisted PK with 403 response
image

### CODE REVIEW

File: cmd/dmsghttp/dmsghttp.go [hunk 1]
```diff
@@ -76,17 +67,33 @@ var rootCmd = &cobra.Command{
...
 		ctx, cancel := cmdutil.SignalContext(context.Background(), log)
 		defer cancel()
+		pk, err := sk.PubKey()
+		if err != nil {
+			pk, sk = cipher.GenerateKeyPair()
+		}
+		if wl != "" {
+			wlk := strings.Split(wl, ",")
+			for _, key := range wlk {
+				var pk1 cipher.PubKey
```
Comment (mrpalide) [inline cmd/dmsghttp/dmsghttp.go:83]: `pk1` make a little confuse

File: cmd/dmsghttp/dmsghttp.go [hunk 2]
```diff
@@ -129,6 +136,57 @@ var rootCmd = &cobra.Command{
...
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	// Check if the remote PK is whitelisted.
+	whitelisted := false
+	if len(wlkeys) == 0 {
+		whitelisted = true
+	} else {
+		for _, pk2 := range wlkeys {
```
Comment (mrpalide) [inline cmd/dmsghttp/dmsghttp.go:154]: `pk2` same as `pk1`

**Review by mrpalide — APPROVED** [review]
Just two little comments.

**Comment by 0pcom** [comment]
will change the variable names on next pr

**Pull request merged**